modules/apps/content-dashboard/content-dashboard-test/src/testIntegration/java/com/liferay/content/dashboard/web/internal/portlet/test/ContentDashboardAdminPortletTest.java

```diff
@@ -574,6 +574,25 @@ public void testGetPropsWithMissingChildAssetVocabulary() throws Exception {
 		}
 	}
 
+	@Test
+	public void testGetRedirect() throws Exception {
+		MVCPortlet mvcPortlet = (MVCPortlet)_portlet;
+
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			_getMockLiferayPortletRenderRequest();
+
+		mvcPortlet.render(
+			mockLiferayPortletRenderRequest,
+			new MockLiferayPortletRenderResponse());
+
+		Assert.assertEquals(
+			"http//localhost/test?",
```

Comment:
Is it not strange the '?' in the end on the URL?

Reply:
indeed! I used the `MockLiferayPortletURL.toString()` as the current URL creating the ThemeDisplay. First I started using s fresh new URL in the `ThemeDisplay` (passing as parameter= but when I saw the `MockLiferayPortletURL` I decided to use it. Another option would be to use a contains instead of an equals. What do you think?

Reply:
With this approach you could have 2 problems:

1. As you are linking your test to the implementation of the class MockLiferayPortletURL, looks like the important part of the test is keep the behaviour of  MockLiferayPortletURL, that I think is not your idea.
2. It is hard to know why this value is MockLiferayPortletURL.toString(), because it is no explained in any part of the test, you need to go to the MockLiferayPortletURL.toString() to know why you assert this value.

Every time that you write a test, you should think about what is the important part you want to assert, and if it is clear for the readers that come after you.

I think that to use a constant value for the current URL is clearer for this purpose, or to assert with MockLiferayPortletURL.toString(), mixing these two options is really confusing.

Thanks!!

Reply:
I totally agree, thanks for the advice ;)